scripts/components/profiles/scatterplot.component.js

```diff
@@ -0,0 +1,181 @@
+  _render() {
+    const margin = {top: 4, right: 6, bottom: 30, left: 29};
+    this.width = this.el.clientWidth - margin.left - margin.right;
+    this.height = 377 - margin.top - margin.bottom;
+    let allYValues = this.data.map(item => item.y);
+    let allXValues = this.data.map(item => item.x[0]);
+
+    this.x = d3_scale_linear()
+      .range([0, this.width])
+      .domain(d3_extent([0, ...allXValues]));
+
+    this.y = d3_scale_linear()
+      .range([this.height, 0])
+      .domain(d3_extent([0, ...allYValues]));
+
+    const xAxis = d3_axis_bottom(this.x)
+      .ticks(8)
+      .tickSize(-this.height, 0)
+      .tickPadding(9)
+      .tickFormat((value, i) => {
+        if (i === 0) {
+          return null;
+        }
+
+        return abbreviateNumber(value, 3);
+      });
+
+    const yAxis = d3_axis_left(this.y)
+      .ticks(7)
+      .tickSize(-this.width, 0)
+      .tickPadding(9)
+      .tickFormat((value) => {
+        return abbreviateNumber(value, 3);
+      });
+
+    this.svg = d3_select(this.el)
+      .append('svg')
+      .attr('width', this.width + margin.left + margin.right)
+      .attr('height', this.height + margin.top + margin.bottom)
+      .append('g')
+      .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+
+    this.svg.append('g')
+      .attr('class', 'axis axis--x')
+      .attr('transform', 'translate(0,' + this.height + ')')
+      .call(xAxis);
+
+    this.svg.append('g')
+      .attr('class', 'axis axis-line')
+      .attr('transform', 'translate(0,' + this.height + ')')
+      .call(d3_axis_bottom(this.x).ticks(0).tickSizeOuter(0));
+
+    this.svg.append('g')
+      .attr('class', 'axis axis--y')
+      .call(yAxis);
+
+    this.svg.append('g')
+      .attr('class', 'axis axis-line')
+      .call(d3_axis_left(this.y).ticks(0).tickSizeOuter(0));
+
+    this.circles = this.svg.selectAll('circle')
+      .data(this._getFormatedData(0))
+      .enter()
+      .append('circle')
+      .attr('class', function(d) { return d.nodeId.toString() === this.nodeId ? 'dot current' : 'dot' }.bind(this))
+      .attr('r', 5)
+      .attr('cx', function(d) { return this.x(d.x); }.bind(this))
+      .attr('cy', function(d) { return this.y(d.y); }.bind(this));
+
+    if (this.showTooltipCallback !== undefined) {
+      this.circles.on('mousemove', function(d) {
+        const selectedSwitcher = document.querySelector('.js-scatterplot-switcher.selected span');
+
+        this.showTooltipCallback(
+          d,
+          {
+            name: selectedSwitcher.getAttribute('data-name'),
+            unit: selectedSwitcher.getAttribute('data-unit'),
+          },
+          d3_event.clientX + 10,
+          d3_event.clientY + window.scrollY + 10
+        );
+      }.bind(this))
+      .on('mouseout', function() {
+        this.hideTooltipCallback();
+      }.bind(this));
+    }
+  }
+
+  _renderXswitcher() {
+    const template = stringToHTML(ScatterplotSwitcherTemplate({data: this.xDimension}));
```

Comment:
We need to stop using `stringToHTML`. We can just do:
```
this.switcherEl.innerHTML = ScatterplotSwitcherTemplate({data: this.xDimension});
```

(provided you created this.switcherEl from an existing DOM element before.)